src/main/java/meldexun/nothirium/mc/renderer/chunk/RenderChunkTaskCompile.java

```diff
@@ -87,35 +91,14 @@ public RenderChunkTaskResult run() {
 					for (int z = 0; z < 16; z++) {
 						pos.setPos(this.renderChunk.getX() + x, this.renderChunk.getY() + y, this.renderChunk.getZ() + z);
 						IBlockState blockState = this.chunkCache.getBlockState(pos);
+						renderBlockState(blockState, pos, visibilityGraph, bufferBuilderPack, mc);
 
-						if (blockState.getRenderType() == EnumBlockRenderType.INVISIBLE) {
-							continue;
-						}
-
-						for (Direction dir : Direction.ALL) {
-							if (blockState.doesSideBlockRendering(chunkCache, pos, EnumFacingUtil.getFacing(dir)))
-								visibilityGraph.setOpaque(pos.getX(), pos.getY(), pos.getZ(), dir);
-						}
-
-						// I will just quote another mod here "This is a ridiculously hacky workaround, I would not recommend it to anyone."
-						blockState.getBlock().hasTileEntity(blockState);
-
-						for (BlockRenderLayer layer : BlockRenderLayerUtil.ALL) {
-							if (Nothirium.isBetterFoliageInstalled ? !BetterFoliage.canRenderBlockInLayer(blockState.getBlock(), blockState, layer) : !blockState.getBlock().canRenderInLayer(blockState, layer)) {
-								continue;
+						if(FluidloggedAPI.isInstalled && FluidloggedUtils.getFluidFromState(blockState) == null) {
+	    					final FluidState fluidState = FluidState.get(pos);
+							if(!fluidState.isEmpty() && (!(blockState.getBlock() instanceof IFluidloggable)
+									|| ((IFluidloggable)blockState.getBlock()).shouldFluidRender(chunkCache, pos, blockState, fluidState))) {
```

Comment:
This will crash if the Fluidlogged API mod is not installed. Classes from the Fluidlogged API mod should only be used in `.../integration/FluidloggedAPI.java`.

Wrong formatting: Missing whitespaces before opening paranthesis of if statement and after closing paranthesis of type casts.